### PR TITLE
fix: expiry date buy now listing

### PR DESCRIPTION
**Pull request by chandni-vault-hill**
## Description

https://github.com/Vault-Hill/vhc-marketplace-app/issues/158

image

Figma Link: https://www.figma.com/file/I1CfJiJLcSv19IcQyPY7Q0/VaultHill?node-id=5768%3A15926

### CODE REVIEW

File: src/features/partials/AssetDetails/index.tsx [hunk 1]
```diff
@@ -281,6 +283,21 @@ const AssetDetails = ({ asset }: { asset: AssetWithListing | undefined }) => {
...
                       />
                     )}
+
+                  {asset?.activeListing?.type === 'BUY_NOW' &&
+                    asset.activeListing.buyNow &&
+                    !asset.activeListing.isComplete && (
+                      <div className='flex flex-row items-center pb-6'>
+                        <InformationCircleIcon className='h-5 w-5 text-[#A96CEF] mr-2' />
+                        <span className='text-[#505780] mr-3'>
+                          Bid active till
```
Comment (creativogee) [inline src/features/partials/AssetDetails/index.tsx:293]: I think the message may be incorrect. Why not "Sale ends on" then "Sale ended on" when time elapse before it is then unmounted when transaction is complete

**Review by creativogee — CHANGES_REQUESTED** [review]

File: src/features/partials/AssetDetails/index.tsx (hunk 1 above)
Comment (chandni-vault-hill) [reply src/features/partials/AssetDetails/index.tsx:293]: Yes, sure thing.

File: src/features/partials/AssetDetails/index.tsx [hunk 2]
```diff
@@ -281,6 +283,24 @@ const AssetDetails = ({ asset }: { asset: AssetWithListing | undefined }) => {
...
                     )}
+
+                  {asset?.activeListing?.type === 'BUY_NOW' &&
+                    asset.activeListing.buyNow &&
+                    !asset.activeListing.isComplete && (
+                      <div className='flex flex-row items-center pb-6'>
+                        <InformationCircleIcon className='h-5 w-5 text-[#A96CEF] mr-2' />
+                        <span className='text-[#505780] mr-3'>
+                          Sale
+                          {new Date(endDate ?? 0).getTime() - Date.now() <= 0
```
Comment (creativogee) [inline src/features/partials/AssetDetails/index.tsx:294]: there is a utility function for this. isDateElapsed()

**Review by creativogee — CHANGES_REQUESTED** [review]

File: src/features/partials/AssetDetails/index.tsx (hunk 2 above)
Comment (chandni-vault-hill) [reply src/features/partials/AssetDetails/index.tsx]: Okay

**Review by creativogee — APPROVED** [review]

**Pull request merged**